e2e-tests/src/localstack.rs

```diff
@@ -0,0 +1,219 @@
+        for queue_url in queue_urls {
+            match sqs_client.delete_queue().queue_url(queue_url).send().await {
+                Ok(_) => log::debug!("Successfully deleted queue: {}", queue_url),
+                Err(e) => eprintln!("Error deleting queue {}: {:?}", queue_url, e),
+            }
+        }
+
+        // Creating SQS queues
+        let mut queue_attributes = HashMap::new();
+        queue_attributes.insert(VisibilityTimeout, "1".into());
+        sqs_client
+            .create_queue()
+            .queue_name(JOB_PROCESSING_QUEUE)
+            .set_attributes(Some(queue_attributes.clone()))
+            .send()
+            .await?;
+        sqs_client
+            .create_queue()
+            .queue_name(JOB_VERIFICATION_QUEUE)
+            .set_attributes(Some(queue_attributes.clone()))
+            .send()
+            .await?;
+        sqs_client
+            .create_queue()
+            .queue_name(JOB_HANDLE_FAILURE_QUEUE)
+            .set_attributes(Some(queue_attributes.clone()))
+            .send()
+            .await?;
+        sqs_client
+            .create_queue()
+            .queue_name(WORKER_TRIGGER_QUEUE)
+            .set_attributes(Some(queue_attributes.clone()))
+            .send()
+            .await?;
+        println!("🌊 SQS queues creation completed.");
+
+        Ok(())
+    }
+
+    /// To set up s3 files needed for e2e testing
+    pub async fn setup_s3(&self) -> color_eyre::Result<()> {
+        let s3_client = self.s3_client().await;
+
+        s3_client.build_test_bucket(&get_env_var_or_panic("AWS_S3_BUCKET_NAME")).await.unwrap();
+
+        // putting the snos output and program output for the given block into localstack s3
+        let snos_output_key = self.l2_block_number.to_string() + "/snos_output.json";
+        let snos_output_json = read("artifacts/snos_output.json").unwrap();
+        s3_client.put_data(Bytes::from(snos_output_json), &snos_output_key).await?;
+        println!("✔️ snos output file uploaded to localstack s3.");
+
+        let program_output_key = self.l2_block_number.to_string() + "/program_output.txt";
+        let program_output = read(format!("artifacts/program_output_{}.txt", self.l2_block_number)).unwrap();
+        s3_client.put_data(Bytes::from(program_output), &program_output_key).await?;
+        println!("✔️ program output file uploaded to localstack s3.");
+
+        // getting the PIE file from s3 bucket using URL provided
+        let file = reqwest::get(format!(
+            "https://madara-orchestrator-sharp-pie.s3.amazonaws.com/{}-SN.zip",
+            self.l2_block_number
+        ))
+        .await?;
+        let file_bytes = file.bytes().await?;
+
+        // putting the pie file into localstack s3
+        let s3_file_key = self.l2_block_number.to_string() + "/pie.zip";
+        s3_client.put_data(file_bytes, &s3_file_key).await?;
+        println!("✔️ PIE file uploaded to localstack s3");
```

Comment:
this part is specific to the e2e test case. imaging that the localstack, mongo etc. files are general for all e2e test cases (if we had more)